utils/spdx/src/test/kotlin/parser/SpdxExpressionLexerTest.kt

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.spdx.parser
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.spec.style.scopes.FunSpecContainerScope
+import io.kotest.datatest.withData
+import io.kotest.matchers.sequences.shouldContainExactly
+import io.kotest.matchers.shouldBe
+
+class SpdxExpressionLexerTest : FunSpec({
+    context("identifiers") {
+        verifyTokens(
+            "a" to sequenceOf(
+                Token.IDENTIFIER(1, 1, "a")
+            ),
+            "a+" to sequenceOf(
```

Comment:
Would an `or-later` also fit in here? (Not sure if it falls under the category of a token)